aten/src/ATen/native/quantized/cpu/qembeddingbag_prepack.cpp

```diff
@@ -101,10 +102,14 @@ c10::intrusive_ptr<EmbeddingPackedParamsBase> PackedEmbeddingBagWeight::prepack(
           for (const auto row : c10::irange(start_idx, end_idx)) {
             const uint8_t* input_row = weight_data + row * embedding_cols;
             std::uint8_t* output_row = output_data + row * output_columns;
-            float* output_row_scale_bias =
-                reinterpret_cast<float*>(output_row + embedding_cols);
-            output_row_scale_bias[0] = weight_scales[row];
-            output_row_scale_bias[1] = weight_bias[row];
+            auto output_row_scale_bias = output_row + embedding_cols;
+            // don't use float* to avoid unaligned address access
+            std::memcpy(
+                output_row_scale_bias, &(weight_scales[row]), sizeof(float));
+            std::memcpy(
+                output_row_scale_bias + sizeof(float),
```

Comment:
is output_row_scale_bias void*? Auto is bit dangerous here because if it is float* then this is not right. Maybe better to explicilty use `void *`

Reply:
It is safe to use auto cast